src/services/portfolioService.js

```diff
@@ -44,8 +78,20 @@ export const getDailyBalances = async () => {
   const labels = [];
   const balances = [];
   const dividends = [];
-  const portfolio = {};
-  let realizedGains = 0;
+  const portfolioHoldings = new PortfolioHoldings();
+  const holdings = portfolioHoldings.holdings;
+
+  // need to account for dates outside of the range
+  const sortedTransactionDates = Object.keys(transactionsByDate).sort();
```

Comment:
@jr094 this part is new, it's to account for any trades that happened before our graph starting point

basically I'm sorting the dates and seeing which ones occur before and update the portfolio dictionary with them first